Assignment6.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class Assignment6 {
+    /**
+     * Rearranges given to make it the same as desired.
+     *
+     * @param given   initial version of the array of numbers.
+     * @param desired final version of the array of numbers.
+     * @return list of moves made during arrangement.
+     */
+    public static List<List<Integer>> rearrange(int[] given, int[] desired) {
+        // Keys are values in given, values are their indexes in given.
+        Map<Integer, Integer> indexMap = new HashMap<>();
+        // Generated for testing purposes.
+        List<List<Integer>> listOfMoves = new LinkedList<>();
+
+        for (int i = 0; i < given.length; i++) {
```

Comment:
Populating the map could also go into a helper method for readability purposes. :)